src/proc/mod.rs

```diff
@@ -313,6 +313,33 @@ impl crate::Expression {
     }
 }
 
+impl crate::Handle<crate::Expression> {
+    /// Return the global variable being accessed by a pointer expression.
+    ///
+    /// Assuming that `pointer` is a series of `Access` and `AccessIndex`
+    /// expressions that ultimately access some part of a `GlobalVariable`,
+    /// return a handle for that global.
+    ///
+    /// If the expression does not ultimately access a global variable, return
+    /// `None`.
+    pub fn originating_global(
+        mut self,
+        func: &crate::Function,
+    ) -> Option<crate::Handle<crate::GlobalVariable>> {
+        loop {
+            self = match func.expressions[self] {
```

Comment:
whaaat, nasty stuff!
Please, let's not add methods with any logic on handles. They are supposed to be just data.
Instead, we can do it with `impl crate::Function`

Reply:
LOL - sorry, I had forgotten about that principle. I'll try to remember it in the future.